Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.py

```diff
@@ -0,0 +1,223 @@
+    :rtype: ``str``
+    """
+    octox.ping()
+    return "ok"
+
+
+def get_adapters(octox: OctoxLabs, *_, **__) -> CommandResults:
+    count, adapters = octox.get_adapters()
+
+    return CommandResults(
+        outputs_prefix="OctoxLabs.Adapters",
+        outputs={
+            "count": count,
+            "results": [
+                convert_to_json(
+                    obj=a,
+                    keys=[
+                        "id",
+                        "name",
+                        "slug",
+                        "description",
+                        "groups",
+                        "beta",
+                        "status",
+                        "hr_status",
+                    ],
+                )
+                for a in adapters
+            ],
+        },
+    )
+
+
+def get_connections(octox: OctoxLabs, args: Dict[str, Any]) -> CommandResults:
+    page = args.get("page", 1)
+    count, connections = octox.get_connections(page=page)
+
+    return CommandResults(
+        outputs_prefix="OctoxLabs.Connections",
+        outputs={
+            "count": count,
+            "results": [
+                convert_to_json(
+                    obj=c,
+                    keys=[
+                        "id",
+                        "adapter_id",
+                        "adapter_name",
+                        "name",
+                        "status",
+                        "description",
+                        "enabled",
+                    ],
+                )
+                for c in connections
+            ],
+        },
+    )
+
+
+def get_discoveries(octox: OctoxLabs, args: Dict[str, Any]) -> CommandResults:
+    page = args.get("page", 1)
+    count, discoveries = octox.get_discoveries(page=page)
+
+    return CommandResults(
+        outputs_prefix="OctoxLabs.Discoveries",
+        outputs={
+            "count": count,
+            "results": [
+                convert_to_json(
+                    d,
+                    keys=[
+                        "id",
+                        "start_time",
+                        "end_time",
+                        "status",
+                        "hr_status",
+                        "progress",
+                    ],
+                )
+                for d in discoveries
+            ],
+        },
+    )
+
+
+def get_last_discovery(octox: OctoxLabs, *_, **__) -> CommandResults:
+    discovery = octox.get_last_discovery()
+    return CommandResults(
+        outputs_prefix="OctoxLabs.Discovery",
+        outputs=convert_to_json(
+            obj=discovery,
+            keys=["id", "start_time", "end_time", "status", "hr_status", "progress"],
+        ),
+    )
+
+
+def search_devices(octox: OctoxLabs, args: Dict[str, Any]) -> CommandResults:
+    fields = args.get("fields", None)
+    if isinstance(fields, str):
+        fields = [f.strip() for f in fields.split(",")]
+
+    count, devices = octox.search_assets(
```

Comment:
what happens if no results are found?

Reply:
<img width="570" alt="image" src="https://user-images.githubusercontent.com/6793206/190366944-38e85ae5-3f06-4ab4-b03d-ed86d08d1888.png">
Return 0 device.

Reply:
Let's replace it with a message: "No results were found for <the query params>"

Reply:
This is not error, it just not found any device with device query. Should I return a message?